TypeCobol.LanguageServer/WorkspaceProjectStore.cs

```diff
@@ -115,51 +103,54 @@ private void AddProject(WorkspaceProject project)
         private WorkspaceProject CreateWorkspaceProject(string projectKey)
         {
             System.Diagnostics.Debug.Assert(projectKey != null);
-            if (_workspaceProjects.ContainsKey(projectKey))
-            {
-                throw new DuplicatedProjectException(projectKey);
-            }
+            
             //We don't have the information about the real RootDirectory.
             //We use the RootDirectory from the Workspace because it has no consequences as long as
             //this folder contains no copy
             CompilationProject compilationProject = new CompilationProject(projectKey, this._workspace.RootDirectory, Helpers.DEFAULT_EXTENSIONS,
                 _workspace.Configuration.Format, this.DefaultWorkspaceProject.Project.CompilationOptions,
                 this.DefaultWorkspaceProject.Project.AnalyzerProvider);
             WorkspaceProject workspaceProject = new WorkspaceProject(projectKey, compilationProject, _workspace);
-            AddProject(workspaceProject);
-            return workspaceProject;
-        }
-
-        /// <summary>
-        /// Refresh all documents in all WorkspaceProject instances
-        /// </summary>
-        internal void RefreshOpenedFiles()
-        {
-            this.DefaultWorkspaceProject.RefreshOpenedFiles();
-            foreach (var wksProject in _workspaceProjects.Values)
+            if (!_namedProjects.TryAdd(projectKey, workspaceProject))
```

Comment:
I think it's better to keep the dedicated method `AddProject(workspaceProject);` here.

Reply:
But I kept the assertion and the exception. `AddProject` is not used elsewhere so I deleted it.